### PR TITLE
rule fixes

### PR DESCRIPTION
**Pull request by jianyexi**
fix 

- [ ] https://github.com/Azure/azure-openapi-validator/issues/404

- [ ] https://github.com/Azure/azure-openapi-validator/issues/440

- [ ] https://github.com/Azure/azure-openapi-validator/issues/439

- [ ] https://github.com/Azure/azure-openapi-validator/issues/438

- [ ] https://github.com/Azure/azure-openapi-validator/issues/437

### CODE REVIEW

**Review by mikekistler — COMMENTED** [review]
This looks good, but I have a couple questions I'd like to resolve before merging.

File: packages/rulesets/src/spectral/functions/Extensions/long-running-response-status-code.ts [hunk 1]
```diff
@@ -1,59 +1,50 @@
...
-const longRunningResponseStatusCode = (
-  methodOp: any,
-  _opts: any,
-  ctx: any,
-  validResponseCodesList: any
-) => {
+const longRunningResponseStatusCode = (methodOp: any, _opts: any, ctx: any, validResponseCodesList: any) => {
   if (methodOp === null || typeof methodOp !== "object") {
-    return [];
+    return []
```
[the remark this answers is not in the document]
Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/Extensions/long-running-response-status-code.ts:5]: We have default prettier config in vscode setting https://github.com/Azure/azure-openapi-validator/blob/856b31b6c6d560e653b6e5fecd9e966c46563cfd/.vscode/settings.json#L8, will add a prettier CI check for it.

File: packages/rulesets/src/spectral/functions/list-in-operation-name.ts [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
+  let gotArray = false
   Object.values(responseList).some((response: any) => {
-    if(response.schema) {
-      if(response.schema.properties?.value?.type === "array")
-      {
-        if(!listRegex.test(swaggerObj['operationId'])){
-          gotArray = true;
-          return true;
+    if (response.schema) {
+      if (response.schema.properties?.value?.type === "array" && Object.keys(response.schema.properties).length === 1) {
```
[the remark this answers is not in the document]
Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/list-in-operation-name.ts:26]: Good feedback, will update the rule.

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**